app/models/user.rb

```diff
@@ -352,13 +352,16 @@ def automator?(track = nil)
 
   def trainer?(track = nil)
     return true if staff?
-    return reputation_for_track(track) >= MIN_REP_TO_TRAIN_ML if track.present?
-
-    # TODO: optimize this
-    User::ReputationPeriod.
-      where(period: :forever, about: :track, user: self).
-      group(:track_id).
-      sum(:reputation).
-      any? { |_, reputation| reputation >= User::MIN_REP_TO_TRAIN_ML }
+    return super() && eligible_for_trainer?(track) if track.present?
+
+    super()
```

Comment:
You could do `return false unless super()` above this to remove the duplicate call from L355 and L357

Reply:
```suggestion
    return false unless super() 
    return eligible_for_trainer?(track) if track.present?

    true
```